python/opae.admin/opae/admin/tools/regmap_debugfs.py

```diff
@@ -0,0 +1,163 @@
+  this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright notice,
+  this list of conditions and the following disclaimer in the documentation
+  and/or other materials provided with the distribution.
+* Neither the name  of Intel Corporation  nor the names of its contributors
+  may be used to  endorse or promote  products derived  from this  software
+  without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+"""
+
+import argparse
+import os.path
+import sys
+
+
+class RegmapDebugfs():
+    """RegmapDebugfs is a class for accessing the debugfs
+       inteface of regmaps in the linux kernel"""
+
+    def __init__(self, regmapdir):
+
+        self.registers_path = os.path.join(regmapdir, 'registers')
+        self.range_path = os.path.join(regmapdir, 'range')
+        self.ranges = []
+        self.record_len = 0
+        self.register_stride = 0
+
+    def get_register_info(self):
+        """get information about regmap"""
+        with open(self.registers_path, 'r') as _file:
+            line0 = _file.readline()
+            line1 = _file.readline()
+            self.record_len = len(line0)
+            list0 = line0.strip().split(': ')
+            list1 = line1.strip().split(': ')
+            self.register_stride = int(list1[0], 16) - int(list0[0], 16)
+
+    def get_range_info(self):
+        """get the list of address ranges for the regmap"""
+        with open(self.range_path, 'r') as _file:
+            for line in _file.readlines():
+                list0 = line.strip().split('-')
+                range_dict = {
+                    'min': int(list0[0], 16),
+                    'max': int(list0[1], 16),
+                    'num_records': ((int(list0[1], 16) - int(list0[0], 16)) /
+                                    self.register_stride) + 1,
+                }
+                self.ranges.append(range_dict)
+
+    def reg_read(self, addr):
+        """reg_read returns the value read from the given
+           address in the regmap"""
+
+        found = False
+        num_records = 0
+        for range_dict in self.ranges:
```

Comment:
You could use here too, the same concept like in the other place.
```
for range_dict in self.ranges:
   (code)
else:
    sys.exit("bad read address")
```

Reply:
I was thinking since I already had an else: with the if, that would be a problem.  I will change to be consistent with the write method.